src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { useState } from 'react';
+
+interface TodoListProps {
+  lodingId: Todo['id'] | null;
+  todos: Todo[];
+  onChange: (todo: Todo, fieldsToUpdate: Partial<Todo>) => Promise<unknown>;
+  onDelete: (todoId: Todo) => void;
+}
+
+export const TodoList: React.FC<TodoListProps> = ({
+  lodingId,
+  todos,
+  onChange,
+  onDelete,
+}) => {
+  const [editId, setEditId] = useState<number | null>(null);
+  const [editValue, setEditValue] = useState('');
+
+  const handleTitleClick = (
+    event: React.MouseEvent<HTMLSpanElement, MouseEvent>,
+    todo: Todo,
+  ) => {
+    if (event.detail === 2) {
+      setEditId(todo.id);
+      setEditValue(todo.title);
+    }
+  };
+
+  const handleSubmit = (todo: Todo) => {
+    onChange(todo, { title: editValue })
+      .then(() => setEditId(null))
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.error(error);
+      });
+  };
+
+  const handleEditSubmit = (
+    event: React.FormEvent<HTMLFormElement>,
+    todo: Todo,
+  ) => {
+    event.preventDefault();
+    handleSubmit(todo);
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <div
+          key={todo.id}
+          data-cy="Todo"
+          className={cn('todo', { completed: todo.completed === true })}
```

Comment:
Use destructuring for `todo` and fix like this
```suggestion
          className={cn('todo', { completed })}
```